sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -591,10 +598,14 @@ ssize_t coap_build_reply_header(coap_pkt_t *pkt, unsigned code,
         }
         *bufpos++ = COAP_PAYLOAD_MARKER;
         *payload = bufpos;
-        hdr_len  = bufpos - (uint8_t *)buf;
         *payload_len_max = len - hdr_len;
```

Comment:
You can move this below the updated `hdr_len`

Reply:
Ah, I now got what you mean. But that would change the behavior: Right now, you can pass `NULL` for `pyaload_len_max` when `payload == NULL`. Moving this outside of the `if (payload)` would no longer allow this. I don't think this is acceptable.

Reply:
How about the latest fixup instead?